client_code/persistence.py

```diff
@@ -49,86 +49,103 @@ def __set__(self, instance, value):
         instance._delta[self._name] = value
 
 
-@classmethod
-def _search(cls, *args, **kwargs):
-    rows = anvil.server.call(f"search_{cls.__name__.lower()}", *args, **kwargs)
-    return (cls.create(store=row) for row in rows)
-
-
-@classmethod
-def _create(cls, store=None, delta=None, *args, **kwargs):
-    instance = cls(*args, **kwargs)
-    instance._store = store or {}
-    instance._delta = delta or {}
-    return instance
+class LinkedClass:
+    "A descriptor class for adding objects based on linked tables as attributes"
 
+    def __init__(self, cls, linked_column=None, *args, **kwargs):
+        self._linked_column = linked_column
+        self._cls = cls
+        self._obj = None
+        self._args = args or []
+        self._kwargs = kwargs or {}
 
-def _getattr(self, key):
-    if self._delta and key in self._delta:
-        return self._delta[key]
+    def __set_name__(self, owner, name):
+        if self._linked_column is None:
+            self._linked_column = name
 
-    return dict(self._store).get(key, None)
+    def __get__(self, instance, objtype=None):
+        if instance is None:
+            return self
 
+        if self._obj is None:
+            self._obj = self._cls(
+                instance._store[self._linked_column], *self._args, **self._kwargs
+            )
 
-def _getitem(self, key):
-    return getattr(self, key)
+        return self._obj
 
+    def __set__(self, instance, value):
+        raise ValueError("Linked Class instance is already set and cannot be changed")
```

Comment:
```suggestion
        raise AttributeError("Linked Class instance is already set and cannot be changed")
```

That's what you get if you try to set a readonly property